internal/proxy/proxy_config.go

```diff
@@ -360,7 +363,9 @@ func parseOptionsConfig(proxy *UpstreamConfig) error {
 	proxy.Timeout = proxy.RouteConfig.Options.Timeout
 	proxy.FlushInterval = proxy.RouteConfig.Options.FlushInterval
 	proxy.HeaderOverrides = proxy.RouteConfig.Options.HeaderOverrides
-	proxy.TLSVerify = proxy.RouteConfig.Options.TLSVerify
+	if proxy.RouteConfig.Options.TLSVerify != nil {
+		proxy.TLSVerify = *proxy.RouteConfig.Options.TLSVerify
+	}
```

Comment:
One item of note, we don't want to set the explicit default value here because of the `if proxy.RouteConfig.Options == nil` short-circuit at the top of the function. Doing so causes half the test suite to fail.